.github/workflows/ci-release.yml

```diff
@@ -43,50 +51,79 @@ jobs:
+#       with:
+#         gpg_private_key: ${{ secrets.GPG_PRIVATE_KEY }}
+#         passphrase: ${{ secrets.GPG_PASSPHRASE }}
+
+#     - name: Build binaries
+#       id: build-binaries
+#       run: make build-all-platforms
+
+#     - name: Package binaries
+#       id: package-binaries
+#       run: bash scripts/package-deploy.sh
+#       if: steps.build-binaries.outcome == 'success'
+
+#     - name: Upload binaries
+#       uses: svenstaro/upload-release-action@2b9d2847a97b04d02ad5c3df2d3a27baa97ce689
+#       with:
+#         file: '{deploy/*.tar.gz,deploy/*.zip,deploy/*.sha256sum.txt,deploy/*.asc}'
+#         file_glob: true
+#         tag: ${{ github.ref }}
+#         repo_token: ${{ secrets.GITHUB_TOKEN }}
+#       if: steps.package-binaries.outcome == 'success'
 
     - uses: docker/setup-qemu-action@e81a89b1732b9c48d79cd809d8d81d79c4647a18
 
     - name: Build and upload all docker images
+      id: build-and-publish
       run: bash scripts/build-upload-docker-images.sh
       env:
         DOCKERHUB_TOKEN: ${{ secrets.DOCKERHUB_TOKEN }}
         QUAY_TOKEN: ${{ secrets.QUAY_TOKEN }}
 
-    - name: Build, test, and publish all-in-one image
-      run: bash scripts/build-all-in-one-image.sh
+    # - name: Build, test, and publish all-in-one image
+    #   run: bash scripts/build-all-in-one-image.sh
+    #   env:
+    #     DOCKERHUB_TOKEN: ${{ secrets.DOCKERHUB_TOKEN }}
+    #     QUAY_TOKEN: ${{ secrets.QUAY_TOKEN }}
+
+    # - name: Build, test, and publish hotrod image
+    #   run: bash scripts/hotrod-integration-test.sh
+    #   env:
+    #     DOCKERHUB_TOKEN: ${{ secrets.DOCKERHUB_TOKEN }}
+    #     QUAY_TOKEN: ${{ secrets.QUAY_TOKEN }}
+
+    - name: Get Image Tag
+      id: tags
+      run: | 
+        echo "tag=$(bash scripts/compute-tags.sh)"
+    
+    - name: Get Image Digest
+      id: digest
+      run: | 
+        echo "imagedigest=$(docker images jaeger-build:${{ steps.tags.outputs.tag }} --digests --quiet --no-trunc)" >> $GITHUB_OUTPUT
+
+    - name: Install Cosign
+      uses: sigstore/cosign-installer@main
+
+    - name: Sign the published Docker image
+          # This step uses the identity token to provision an ephemeral certificate
+          # against the sigstore community Fulcio instance.
       env:
-        DOCKERHUB_TOKEN: ${{ secrets.DOCKERHUB_TOKEN }}
-        QUAY_TOKEN: ${{ secrets.QUAY_TOKEN }}
+        DIGEST: ${{ steps.build-and-publish.outputs.digest }}
+        TAG: ${{ steps.tags.outputs.tag }}
+      run: echo "${TAG}, ${DIGEST}"
+      # echo "${TAG}" | xargs -I {} cosign sign --yes {}@${DIGEST}
 
-    - name: Build, test, and publish hotrod image
-      run: bash scripts/hotrod-integration-test.sh
+    - name: Sign the Container Images
       env:
-        DOCKERHUB_TOKEN: ${{ secrets.DOCKERHUB_TOKEN }}
-        QUAY_TOKEN: ${{ secrets.QUAY_TOKEN }}
+        COSIGN_EXPERIMENTAL: "true"
+      run: |
+        cosign sign jaeger-build@${{ steps.digest.outputs.imagedigest }}
```

Comment:
you can read more about the risks of script injections: [docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-an-intermediate-environment-variable](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-an-intermediate-environment-variable)

Reply:
Any suggestion plz! `cosign sign jaeger-build@${{ steps.digest.outputs.imagedigest }}` didn't work for me.